### PR TITLE
system variables: add `tidb_analyze_version`

### PR DESCRIPTION
**Pull request by winoros**
Signed-off-by: Yiding Cui <winoros@gmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add the variable which is added in #6410 

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -206,6 +206,12 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 默认值：OFF
 - 这个变量用来控制是否允许通过 `ALTER TABLE MODIFY` 或 `ALTER TABLE CHANGE` 来移除某个列的 `AUTO_INCREMENT` 属性。默认 (`OFF`) 为不允许。
 
+### `tidb_analyze_version` <span class="version-mark">从 v5.1.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值: 2
+- 这个变量用于控制 tidb 收集和使用统计信息的行为，目前的取值范围是 1 和 2，具体的行为可以参照[统计信息简介](statistics.md)。当前值为 2 时的行为还是实验性特性。
```
Comment (time-and-fate) [inline system-variables.md:213]:
```suggestion
- 这个变量用于控制 TiDB 收集和使用统计信息的行为，目前的取值范围是 1 和 2，具体的行为可以参照[统计信息简介](statistics.md)。当前值为 2 时的行为还是实验性特性。
```

**Comment by TomShawn** [comment]
@winoros 从这里可知，其实 v5.0 也有这个变量，是否需要 cherry-pick 到 5.0 呢？
image

File: system-variables.md [hunk 2]
```diff
@@ -206,6 +206,12 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 默认值：OFF
 - 这个变量用来控制是否允许通过 `ALTER TABLE MODIFY` 或 `ALTER TABLE CHANGE` 来移除某个列的 `AUTO_INCREMENT` 属性。默认 (`OFF`) 为不允许。
 
+### `tidb_analyze_version` <span class="version-mark">从 v5.1.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值: 2
+- 这个变量用于控制 TiDB 收集统计信息的行为，目前的取值范围是 1 和 2，具体的行为可以参照[统计信息简介](statistics.md)。当前值为 2 时的行为还是实验性特性。
```
Comment (TomShawn) [inline system-variables.md:213]:
```suggestion
- 这个变量用于控制 TiDB 收集统计信息的行为。
- 可选值：1 和 2
- 默认值：2
- 在 v5.1.0 以前的版本中，该变量的默认值为 `1`。在 v5.1.0 中，该变量的默认值为 `2`，作为实验特性使用。具体可参照[统计信息简介](/statistics.md)文档。
```

File: system-variables.md [hunk 3]
```diff
@@ -206,6 +206,15 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 默认值：OFF
 - 这个变量用来控制是否允许通过 `ALTER TABLE MODIFY` 或 `ALTER TABLE CHANGE` 来移除某个列的 `AUTO_INCREMENT` 属性。默认 (`OFF`) 为不允许。
 
+### `tidb_analyze_version` <span class="version-mark">从 v5.1.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值: 2
```
Comment (TomShawn) [inline system-variables.md:212]:
```suggestion
```

File: system-variables.md [hunk 4]
```diff
@@ -206,6 +206,15 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+- 这个变量用于控制 TiDB 收集统计信息的行为。
+- 可选值：1 和 2
+- 默认值：2
```
Comment (TomShawn) [inline system-variables.md:215]:
```suggestion
- 作用域：SESSION | GLOBAL
- 可选值：1 和 2
- 默认值：2
- 这个变量用于控制 TiDB 收集统计信息的行为。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**